### ISSUE TITLE
What's in result.txt?

### ISSUE THREAD

**Issue by TommyZihao** [ultralytics/yolov5#260]
Is it training metrics or validation metrics?
![image](https://user-images.githubusercontent.com/36354458/86265357-4314b180-bbf6-11ea-8878-963fd84ad9d4.png)

**Comment by TommyZihao** [comment]
Epoch   gpu_mem      GIoU       obj       cls     total   targets  img_size Class      Images     Targets           P           R      mAP@.5  mAP@.5:.95
Which is training metrics and which is validation metrics?

**Comment by glenn-jocher** [comment]
@TommyZihao last 3 columns are validation losses (giou obj cls). You can plot your results.txt file like this:

```python
from utils.utils import *
plot_results_overlay()
```

**Comment by Zpadger** [comment]
It should be "Epoch   gpu_mem    GIoU      Obj       Cls      total       targets  img_size    P         R      mAP@.5  mAP@.5:.95 val_GIoU   val_obj  val_cls"

**Comment by Yu-Hang** [comment]
> @TommyZihao last 3 columns are validation losses (giou obj cls). You can plot your results.txt file like this:
> 
> ```python
> from utils.utils import *
> plot_results_overlay()
> ```

`Epoch gpu_mem GIoU Obj Cls total targets img_size P R mAP@.5 mAP@.5:.95 val_GIoU val_obj val_cls`

@glenn-jocher So, the results before the last 3 columns - P R mAP@.5 mAP@.5:.95 - these are all training metrics?

**Comment by glenn-jocher** [comment]
@Yu-Hang no, the last 3 columns are validation losses. There's not much point in reading your results.txt file, I would recommend you simply plot it:

Training results are automatically logged to [Tensorboard](https://www.tensorflow.org/tensorboard) and `runs/train/exp/results.txt`, which is plotted as `results.png` (below) after training completes. You can also plot any `results.txt` file manually:

```python
from utils.plots import plot_results 
plot_results(save_dir='runs/train/exp')  # plot all results*.txt files in 'runs/train/exp'
```

<p align="left"><img width="800" alt="COCO128 Training Results" src="https://github.com/ultralytics/yolov5/releases/download/v1.0/results_COCO128.png"></p>

**Comment by DeoLukamba** [comment]
> @TommyZihao last 3 columns are validation losses (giou obj cls). You can plot your results.txt file like this:
> 
> ```python
> from utils.utils import *
> plot_results_overlay()
> ```

what does it mean cls of 0.462? Is cls integer or float?  I have cls of 0 , is that correct? please looking for your reply
![results](https://user-images.githubusercontent.com/77050666/138309672-038a7143-cbf5-4dda-8ecf-7e6772f06f4c.JPG)

**Comment by glenn-jocher** [comment]
@DeoLukamba cls is classification loss. Losses are all floats. Single-class datasets will have 0 cls loss.

**Comment by xingjianzhi** [comment]
> > @TommyZihao last 3 columns are validation losses (giou obj cls). You can plot your results.txt file like this:
> > ```python
> > from utils.utils import *
> > plot_results_overlay()
> > ```
> 
> what does it mean cls of 0.462? Is cls integer or float? I have cls of 0 , is that correct? please looking for your reply ![results](https://user-images.githubusercontent.com/77050666/138309672-038a7143-cbf5-4dda-8ecf-7e6772f06f4c.JPG)

what do you train? the training result are so better

**Comment by glenn-jocher** [comment]
@xingjianzhi I do not train a specific dataset myself. The results may vary depending on the dataset, size, and environment used for training. Generally, for better training results, it's recommended to use larger and more diverse datasets, along with proper hyperparameter tuning and training techniques like data augmentation and transfer learning.
